crates/rust-eigenda-v2-client/README.md

```diff
@@ -2,9 +2,113 @@
 
 Rust client for interacting with EigenDA V2.
 
+[More about V2 implementation](https://docs.eigenda.xyz/releases/blazar).
+
+## Sample usage
+
+### Dispersal and Retrieval
+
+This example uses the Holesky ETH chain.
+
+```rs
+use std::str::FromStr;
+
+use ethereum_types::H160;
+use rust_eigenda_signers::signers::private_key::Signer;
+use rust_eigenda_v2_client::{
+    payload_disperser::{PayloadDisperser, PayloadDisperserConfig},
+    relay_client::{RelayClient, RelayClientConfig},
+    relay_payload_retriever::{RelayPayloadRetriever, RelayPayloadRetrieverConfig, SRSConfig},
+    utils::SecretUrl,
+};
+use rust_eigenda_v2_common::{EigenDACert, Payload, PayloadForm};
+
+const CERT_VERIFIER_ADDRESS: H160 = H160([
+    0xfe, 0x52, 0xfe, 0x19, 0x40, 0x85, 0x8d, 0xcb, 0x6e, 0x12, 0x15, 0x3e, 0x21, 0x04, 0xad, 0x0f,
+    0xdf, 0xbe, 0x11, 0x62,
+]);
+const HOLESKY_RELAY_REGISTRY_ADDRESS: H160 = H160([
+    0xac, 0x8c, 0x6c, 0x7e, 0xe7, 0x57, 0x29, 0x75, 0x45, 0x4e, 0x2f, 0x0b, 0x5c, 0x72, 0x0f, 0x9e,
+    0x74, 0x98, 0x92, 0x54,
+]);
+const HOLESKY_ETH_RPC_URL: &str = "https://ethereum-holesky-rpc.publicnode.com";
+const HOLESKY_DISPERSER_RPC_URL: &str = "https://disperser-testnet-holesky.eigenda.xyz";
+// This private key won't work, you need to set up your own with a set
+const ACCOUNTANT_PRIVATE_KEY: &str = "<YOUR_PRIVATE_KEY>";
+// You need to have a g1 point downloaded locally for this example
+const G1_POINT_FILE_PATH: &str = "../resources/g1.point";
+
+#[tokio::main]
+async fn main() {
+    let data = vec![42];
+    let eigenda_cert = disperse(data.clone()).await;
+    let payload = retrieve(eigenda_cert).await;
+    assert_eq!(data, payload);
+}
+
+async fn disperse(data: Vec<u8>) -> EigenDACert {
```

Comment:
after a second thought, I think it will be clearer to move the configs to [LazyLocks](https://doc.rust-lang.org/beta/std/sync/struct.LazyLock.html) (have them defined at the top-level instead of in a auxiliary function.

After doing that change you can remove the `disperse` and `retrieve` functions and just calls the relevant methods.